### DISCUSSION TITLE
Configuring ejabberd Sip mod with external authentication

### DISCUSSION THREAD

**Discussion by Fabiokleis** [processone/ejabberd#4204] [Questions]
I'm trying to configure an ejabberd sip server using external authentication script.

My problem happens when I try to send REGISTER message to sip server from Linphone desktop, ejabberd always reply 401 unauthorized and the external script never are called, I don't know why cuz when I tested locally by running `ejabberdctl check_password` it worked as expected, I can see the external script logs.

How I can trace the problem, ejabberd validates something before calling external script that I have missed? maybe Linphone desktop creates an invalid message?

**Comment by badlop** [comment]
Hello, it seems nobody has experience enough in SIP to answer you. Did you solve the problem somehow, or maybe simply dropped the idea?

Some possibilities to try:
- Use the default ejabberd authentication. Does client auth work that way?
- Setup loglevel option to `debug`. Maybe you are lucky and it shows some meaningful message...
- Provide the minimum reproduction steps, so somebody else can try to reproduce that problem, and maybe investigate it themselves.
